jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1408,12 +1406,14 @@ public CompletableFuture<Void> writeInterim(int status, HttpFields headers)
                 if (status == HttpStatus.CONTINUE_100)
                 {
                     if (!httpChannelState._expects100Continue)
-                        return UNEXPECTED_100_CONTINUE;
+                        return CompletableFuture.failedFuture(new IllegalStateException("100 not expected"));
+                    if (_request.getLength() == 0)
```

Comment:
This is not a sufficient test to check if a request has content.  Need to also check content-encoding at least.  See https://datatracker.ietf.org/doc/html/rfc9112#name-message-body-length.   Like `hasContent` in `AbstractProxyServlet`

Also would it not be better to not set `_expects100Continue` to `true` if there is no content?

Reply:
@sbordet Perhaps to the check for a body in org.eclipse.jetty.http.MetaData.Request#is100ContinueExpected or org.eclipse.jetty.server.internal.HttpChannelState#onRequest?

Reply:
Oh I think this is OK now because it is the inverse!